### ISSUE TITLE
Define vs Inline

### ISSUE THREAD

**Issue by OdnetninI** [OdnetninI/CLib#1]
Must change syscall macro to inline function?

**Comment by OdnetninI** [comment]
Inline version looks nicer
```
#define asm __asm__

#define syscall(...) do{ syscall_get( __VA_ARGS__, syscall_2, syscall_1 )( __VA_ARGS__ ); } while(0)
#define syscall_get(_1, _2, name, ...) name

#define NO_RETURN while(1)

inline void syscall_1(int eax) {
    asm("int $0x80" : : "a" (eax));
}

inline void syscall_2(int eax, int ebx) {
    asm("int $0x80" : : "a" (eax), "b" (ebx));
}

__attribute__((noreturn)) void sys_exit(int code) {
    syscall(1, code);
    NO_RETURN;
}
```

```
sys_exit(int):
        push    rbx
        mov     eax, 1
        mov     ebx, edi
        int $0x80
.L2:
        jmp     .L2 
```

It's almost the same as macro version:
```
#define asm __asm__
#define NO_RETURN while(1)

/* Auto-define SYSCALLS (Not Return Value) */
#define SYSCALL_NOTR_(...) do{ SYSCALL_NOTR_get( __VA_ARGS__, SYSCALL_NOTR_2, SYSCALL_NOTR_1 )( __VA_ARGS__ ); } while(0)
#define SYSCALL_NOTR_get(_1, _2, name, ...) name

#define SYSCALL_NOTR_1(syscall) asm ("int $0x80" : : "a" (syscall))
#define SYSCALL_NOTR_2(syscall, value) asm ("int $0x80" : : "a" (syscall), "b" (value))

/* Auto-generate functions containers */
#define SYSCALL_NOTR(name, ...) void syscall_##name (__VA_ARGS__)


__attribute__((noreturn)) SYSCALL_NOTR(exit, int value);

__attribute__((noreturn)) SYSCALL_NOTR(exit, int value) {
  SYSCALL_NOTR_(1, value);
  NO_RETURN;
}
```

```
syscall_exit(int):
        push    rbx
        mov     eax, 1
        mov     ebx, edi
        int $0x80
.L2:
        jmp     .L2
```

**Comment by OdnetninI** [comment]
Must add halt before loop for exit syscall?

**Comment by OdnetninI** [comment]
I'm wasting time on this,
Is really needed?

I can user inline asm for every syscall, so theres no ansi c inline errors and less overheading.

**Comment by OdnetninI** [comment]
Solved: 1d92bd51483917920e3c5965a49fdcd1965f703a
